src/NuGet.Core/NuGet.Commands/RestoreCommand/Diagnostics/UnresolvedMessages.cs

```diff
@@ -20,21 +20,20 @@ namespace NuGet.Commands
     /// <summary>
     /// Log errors for packages and projects that were missing.
     /// </summary>
-    public static class UnresolvedMessages
+    internal static class UnresolvedMessages
     {
         /// <summary>
         /// Log errors for missing dependencies.
         /// </summary>
-        public static async Task LogAsync(IEnumerable<IRestoreTargetGraph> graphs, RemoteWalkContext context, ILogger logger, CancellationToken token)
+        internal static async Task LogAsync(IEnumerable<IRestoreTargetGraph> graphs, RemoteWalkContext context, CancellationToken token)
         {
-            var tasks = graphs.SelectMany(graph => graph.Unresolved.Select(e => GetMessageAsync(graph.TargetGraphName, e, context.RemoteLibraryProviders, context.CacheContext, logger, token))).ToArray();
+            var tasks = graphs.SelectMany(graph => graph.Unresolved.Select(e => GetMessageAsync(graph.TargetGraphName, e, context.FilterDependencyProvidersForLibrary(e), context.CacheContext, context.Logger, token))).ToArray();
```

Comment:
Whenever a package is unresolved during restore, the following error messages are logged. Hence `context.FilterDependencyProvidersForLibrary(e)` is again invoked to filter sources based on package namespaces configuration.

https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1101
https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1102
https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1103

Let's take NU1102 example if a package is unresolved the following message will be logged without package namespaces configuration.

Unable to find package 'PackageId' with version (>= 9.0.1)
  - Found 30 version(s) in 'sourceA' [ Nearest version: '4.0.0' ]
  - Found 10 version(s) in 'sourceB' [ Nearest version: '4.0.0-rc-2129' ]
  - Found 9 version(s) in 'sourceC' [ Nearest version: '3.0.0-beta-00032' ]
  - Found 0 version(s) in 'sourceD'
  - Found 0 version(s) in 'sourceE'


Assume we have package namespaces configuration that says for package 'PackageId' only look in `sourceA` , `sourceC`. In this case the error message should be as follows. A call to `context.FilterDependencyProvidersForLibrary(e)` filters package sources as per namespaces configuration.

Unable to find package 'PackageId' with version (>= 9.0.1)
  - Found 30 version(s) in 'sourceA' [ Nearest version: '4.0.0' ]
  - Found 9 version(s) in 'sourceC' [ Nearest version: '3.0.0-beta-00032' ]

Reply:
I really appreciate these comments that are meant to make reviewing easier. 
I particuarly appreciate the error/warning examples.

Should we call  out that the other sources were unused due to a namespaces configuration?
I want this to be a start of a discussion, I think it's perfectly reasonable to address in a follow up, if any changes were deemed necessary.

Reply:
btw, I'm aware that there are debug logs with that information, but often times people will only see the error message.

Reply:
>Should we call out that the other sources were unused due to a namespaces configuration?

I like this idea. The extra message at the end `The sources sourceB, sourceD and sourceE are skipped because of the package namespaces configuration. ` will be actionable for customers

```
Unable to find package 'PackageId' with version (>= 9.0.1)
  - Found 30 version(s) in 'sourceA' [ Nearest version: '4.0.0' ]
  - Found 9 version(s) in 'sourceC' [ Nearest version: '3.0.0-beta-00032' ]

The sources sourceB, sourceD and sourceE are skipped because of the package namespaces configuration. 
```
@chgill-MSFT thoughts?

Reply:
Please note I already included very similar message if package namespace filter applied during search or not, it only logs once for debug logs.

https://github.com/NuGet/NuGet.Client/blob/63b8dbad9399c4eaade9a2ccdecee7056a69bc7a/src/NuGet.Core/NuGet.PackageManagement/PackageDownloader.cs#L106-L116

Reply:
Logging message for each source(M) is ruled out because sometimes hundreds of packages are restored once. 
Then it it'll log N(package count)*M(source count count) messages.